zcash_client_sqlite/src/wallet.rs

```diff
@@ -2292,13 +2385,86 @@ pub(crate) fn put_tx_data(
         ":expiry_height": u32::from(tx.expiry_height()),
         ":raw": raw_tx,
         ":fee": fee.map(u64::from),
+        ":target_height": target_height.map(u32::from),
     ];
 
     stmt_upsert_tx_data
-        .query_row(tx_params, |row| row.get::<_, i64>(0))
+        .query_row(tx_params, |row| row.get::<_, i64>(0).map(TxRef))
         .map_err(SqliteClientError::from)
 }
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub(crate) enum TxQueryType {
+    Enhancement,
+    Status,
```

Comment:
```suggestion
    Status,
    Enhancement,
```
to follow the order of the `match`es below.